benches/nexmark/generator/config.rs

```diff
@@ -27,26 +34,52 @@ pub struct Config {
 /// Implementation of config methods based on the Java implementation at
 /// [GeneratorConfig.java](https://github.com/nexmark/nexmark/blob/v0.2.0/nexmark-flink/src/main/java/com/github/nexmark/flink/generator/GeneratorConfig.java)
 impl Config {
-    pub fn new(nexmark_config: NexmarkConfig, base_time: u64, first_event_number: usize) -> Config {
+    pub fn new(
+        nexmark_config: NexmarkConfig,
+        base_time: u64,
+        first_event_id: u64,
+        max_events_or_zero: u64,
+        first_event_number: usize,
+    ) -> Config {
         let inter_event_delay =
             1_000_000 / nexmark_config.first_event_rate * nexmark_config.num_event_generators;
 
+        // Original Java implementation says:
+        // "Scale maximum down to avoid overflow in getEstimatedSizeBytes."
+        // but including to ensure similar behavior.
+        let max_events = match max_events_or_zero {
+            0 => {
+                u64::MAX
+                    / (nexmark_config.total_proportion() as u64
+                        * [
+                            nexmark_config.avg_person_byte_size as u64,
+                            nexmark_config.avg_auction_byte_size as u64,
+                            nexmark_config.avg_bid_byte_size as u64,
+                        ]
+                        .iter()
+                        .max()
+                        .unwrap())
+            }
```

Comment:
```suggestion
            0 => {
                let max_average = [
                    nexmark_config.avg_person_byte_size as u64,
                    nexmark_config.avg_auction_byte_size as u64,
                    nexmark_config.avg_bid_byte_size as u64,
                ]
                .iter()
                .max()
                .unwrap();
                u64::MAX / (nexmark_config.total_proportion() as u64 * max_average)
            }
```

Reply:
Updated, but ended up being slightly different, needing to deref `max_average` (as `.max()` returns an `Option<&usize>` here), and without the deref, the borrow-checker complains: 

```
error[E0716]: temporary value dropped while borrowed
  --> benches/nexmark/generator/config.rs:52:35
   |
52 |                   let max_average = [
   |  ___________________________________^
53 | |                     nexmark_config.avg_person_byte_size as u64,
54 | |                     nexmark_config.avg_auction_byte_size as u64,
55 | |                     nexmark_config.avg_bid_byte_size as u64,
56 | |                 ]
   | |_________________^ creates a temporary which is freed while still in use
...
59 |                   .unwrap();
   |                            - temporary value is freed at the end of this statement
60 |                   u64::MAX / (nexmark_config.total_proportion() as u64 * max_average)
   |                                                                          ----------- borrow later used here
   |
   = note: consider using a `let` binding to create a longer lived value

```

though it's not clear to me why temporary value hasn't already been used to calculate the max_average referenced value when used. Anyway, also then moved the cast to u64 so it's just done once for `max_average`.